### PR TITLE
Refactor the notary command line to not use global mutable state

### PR DESCRIPTION
**Pull request by cyli**
… and to not exit on error.

I couldn't quite debug the flags persisting between tests while I was working on #452 - this should make it easier to test. (and seems to fix my problems for that issue)

To that end, I added a couple tests.

### CODE REVIEW

**Comment by diogomonica** [comment]
@cyli rebase needed.

**Comment by diogomonica** [comment]
Really good refactor. This is a lot more consistent now. Thank you for the extra tests!

**Comment by cyli** [comment]
Sorry, https://github.com/docker/notary/commit/0f77d1a318a76415ff37f5b00c3b486f26f74bf8#diff-5d667de3427093c765c7dc4442dbf926R16 was what was failing on the CI.  The default location is hard to test :|

**Comment by cyli** [comment]
Nope, actually, it looks like it was just looking in the cwd, and I think that was because it was just synthesizing it from the config ext type and config file name.  I'm just setting the config file outright instead, and that seems to fix it.

**Comment by diogomonica** [comment]
@cyli so this is ready to be merged?

File: cmd/notary/main_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	"key restore tempfile.zip",
+	"key import backup.pem",
+	"key remove e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+	"key passwd e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+	"cert list",
+	"cert remove e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+	"delegation list repo",
+	"delegation add repo targets/releases path/to/pem/file.pem",
+	"delegation remove repo targets/releases",
+}
```
Comment (riyazdf) [inline cmd/notary/main_test.go]: this is an awesome test!!

**Comment by riyazdf** [comment]
LGTM, thanks so much for cleaning this up :+1:

**Comment by cyli** [comment]
@diogomonica yep!

**Pull request merged**